zerver/views/tusd.py

```diff
@@ -0,0 +1,100 @@
+import os
+import random
+from shutil import move
+from typing import Any, Dict, Mapping
+
+from django.conf import settings
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import gettext as _
+
+from zerver.decorator import internal_notify_view
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.lib.upload import check_upload_within_quota
+from zerver.lib.upload.base import create_attachment, sanitize_name
+from zerver.lib.upload.s3 import S3UploadBackend
+from zerver.models import UserProfile
+
+
+def generate_path(realm_id: str, file_id: str, uploaded_file_name: str) -> str:
+    return "/".join(
+        [
+            realm_id,
+            format(random.randint(0, 255), "x"),
```

Comment:
Later if we remove `file_id` field from Attachment, than we will have only file's parent folder with that ID so we can identify using `file_id`. If we don't need this than I will use already present `generate_message_upload_path` method.

Reply:
If we remove `file_id` from Attachment, it will be because we are using tusd v2 -- in which case we can control where the file is placed as well as the return code to the client, so tusd's internal `file_id`  will be completely irrelevant to us.  The filepaths we write to live forever, even after we upgrade, so why bund ourselves to putting `file_id` in them?

Reply:
Now we are using the `generate_message_upload_path` method to generate the path.